x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapUtils.java

```diff
@@ -187,7 +198,8 @@ protected void doRun() throws Exception {
 
             @Override
             public void onFailure(Exception e) {
-                LOGGER.debug("LDAP bind [{}] failed for [{}] - [{}]", bind, ldap, e.toString());
+                Level logLevel = systemBind ? Level.WARN : Level.DEBUG;
```

Comment:
There might be situations where the warning messages could get excessive? But I don't like throttle for the type of message either. Also, is it viable to wrap a similar message back to user, e.g. `runnable.onFailure(new ElasticsearchSecurityException("LDAP bind ...", e, ...)`?

Reply:
The issue is that this method (`maybeForkThenBind`) routes all exceptions through the same handler. But not all exceptions are equal, and only the caller really knows which ones should be warnings.
However, reworking it to have separate exception handlers for the `bind` vs the `fork` would end up being a bigger change than I could justify, just to adjust some logging levels.

We _could_ wrap the LDAP exception here, but I'm not sure whether that's better or worse. It potentially means that the exception itself has less information, and the useful details are in the cause.
Understanding the impacts of that change is greater than simply adjusting some logging levels.